dist/20.10.sh

```diff
@@ -0,0 +1,769 @@
+#!/bin/sh
+set -e
+# Docker CE for Linux installation script
+#
+# See https://docs.docker.com/engine/install/ for the installation steps.
+#
+# This script is meant for quick & easy install via:
+#   $ curl -fsSL https://get.docker.com -o get-docker.sh
+#   $ sh get-docker.sh
+#
+# For test builds (ie. release candidates):
+#   $ curl -fsSL https://test.docker.com -o test-docker.sh
+#   $ sh test-docker.sh
+#
+# NOTE: Make sure to verify the contents of the script
+#       you downloaded matches the contents of install.sh
+#       located at https://github.com/docker/docker-install
+#       before executing.
+#
+# Git commit from https://github.com/docker/docker-install when
+# the script was uploaded (Should only be modified by upload job):
+SCRIPT_COMMIT_SHA="93d2499759296ac1f9c510605fef85052a2c32be"
+
+CHANNEL="stable"
+DOWNLOAD_URL="https://download.docker.com"
+REPO_FILE="docker-ce.repo"
+temp_version=$(find . -maxdepth 1 -name '20.10.*' | sort -V | tail -1)
```

Comment:
This is what we are distributing, there is no list of versions available for the node that is installing this? Is this supposed to be in the dist script, and if yes, how does this determine the latest version if the repo content is not available at `.`? Or are there other instructions needed for this?

Reply:
Actually the process here is, if someone try to install docker using `https://releases.rancher.com/install-docker/20.10.sh | sh` it will try to find the latest available docker version which are inside `dist/` (assuming all the versions are inside `dist/` [see this](https://github.com/rancher/install-docker/blob/881230f9dc7e6d8d9bde29d3250b2285018e242e/scripts/sync?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L4)) because these are distributing by Rancher, so this will try to install the latest version available in `dist/`. And then it will pick that version and try to install it on the node. If that version is not supported on Node OS then the script will pick the second latest version from `dist/` and try to install it and so on.
As we are adding every docker versions here after an upstream release, it shouldn't be a problem I think.